src/components/popup/home/Balance.tsx

```diff
@@ -37,6 +36,45 @@ export default function Balance() {
     false
   );
 
+  useEffect(() => {
+    // TODO: The balance and fiat balance should be loaded and calculated from a provider / background service. Relying
```

Comment:
See here. Right now we have many components handling business logic, which causes issues like this where the functionality / data we need is inside a component, which means that it's only available when that component is rendered, and that in order to "subscribe" to that data we end up adding logic inside the component that doesn't belong there (separation of concerns, single responsibility). 

This is ok-ish for now, but once we have proper state management with Mobx, things like this should be moved there.